entry.go

```diff
@@ -82,7 +82,10 @@ func (entry *Entry) WithFields(fields Fields) *Entry {
 	for k, v := range fields {
 		data[k] = v
 	}
-	return &Entry{Logger: entry.Logger, Data: data}
+	// shallow copy
+	newentry := *entry
+	newentry.Data = data
+	return &newentry
```

Comment:
It *seems* to be safe: so far the implementation of `Entry.log()` overwrites `Entry.Buffer` [every time](https://github.com/sirupsen/logrus/blob/master/entry.go#L104) and the buffer pool [is protected](https://github.com/sirupsen/logrus/blob/master/entry.go#L11) with `sync.Pool`.